src/inference/src/ie_core.cpp

```diff
@@ -1291,12 +1331,12 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
      *        Such extensions can be used for both CNNNetwork readers and device plugins
      */
     void AddExtension(const ie::IExtensionPtr& extension) {
-        std::lock_guard<std::mutex> lock(pluginsMutex);
+        std::lock_guard<std::mutex> lock(get_mutex());
         AddExtensionUnsafe(extension);
```

Comment:
in `AddExtensionUnsafe` we change all the plugins and they are unsafe. So, `AddExtension` is unsafe as well

Reply:
can we document at the beginning where we define mutexes - what mutex is responsible for what. In this case we can understand whether we have unsafe critical sections